extensions/amp-consent/0.1/consent-config.js

```diff
@@ -150,7 +152,40 @@ export class ConsentConfig {
       }
     }
 
-    return config;
+    return this.mergeGeoOverride_(config).then(() => config);
+  }
+
+  /**
+   * Merge correct geoOverride object into toplevel config, then
+   * validate.
+   * @param {JsonObject} config
+   * @return {!Promise}
+   */
+  mergeGeoOverride_(config) {
+    if (config['geoOverride']) {
+      Services.geoForDocOrNull(this.element_).then(geoService => {
+        userAssert(geoService, 'requires <amp-geo> to use geoOverride');
+        const geoGroups = Object.keys(config['geoOverride']);
+        for (let i = 0; i < geoGroups.length; i++) {
+          if (geoService.isInCountryGroup(geoGroups[i]) === GEO_IN_GROUP.IN) {
+            config = /** @type {!JsonObject} */ (deepMerge(
```

Comment:
let's not recursively merge all the geoGroups that matches. the spec said the geoGroups should be mutually exclusive.

it's safe for us to just merge the first one.